js-arrays/arrays.js

```diff
@@ -0,0 +1,187 @@
+function isArrayOne(a) {
+  return a instanceof Array;
+}
+
+function isArrayTwo(a) {
+  return Array.isArray(a);
+}
+
+function isArrayThree(a) {
+  return Object.prototype.toString.call(a) === "[object Array]";
+}
+
+function range(a, b, c) {
+  var arr = [];
+  if (b === undefined || b === null) {
+    if (c !== undefined) {
+      for (var i = 0; i < a; i += c) {
+        arr.push(i);
+      }
+    } else {
+      for (var i = 0; i < a; i++) {
+        arr.push(i);
+      }
+    }
+  } else if (c === undefined) {
+    for (var i = a; i < b; i++) {
+      arr.push(i);
+    }
+  } else if (c !== 0) {
+    for (var i = a; i < b; i += c) {
+      arr.push(i);
+    }
+  }
+  return arr;
+}
+
+function compactLoop(a) {
+  var arr = [];
+  if (Array.isArray(a)) {
+    for (var i = 0; i < a.length; i++) {
+      if (a[i]) {
+        arr.push(a[i]);
+      }
+    }
+    return arr;
+  } else {
+    console.log(a + " is not an array");
+  }
+}
+
+function compact(a) {
+  if (Array.isArray(a)) {
+    var arr = a.filter(function(el) {
+      return el ? true : false; 
```

Comment:
In line 41 I check if element of the array is truthy or not and push it if it does into a new array. In the line 54 it checks if the element of the array is truthy too, but return `true` or `false` to the filter and it keeps the element in the new array if it's `true`